python/triqs_cthyb/solver.py

```diff
@@ -145,11 +148,38 @@ def solve(self, **params_kw):
         # (only supported for G_tau, to permit compatibility with dft_tools)
         if perform_post_proc and (self.last_solve_parameters["measure_G_tau"] == True):
 
+            if self.last_solve_parameters["measure_density_matrix"]:
```

Comment:
It would be nice if the calculated quantities were otherwise set to `None`, what do you think?

That makes it easier for the user and ourselves to check if they have been set

```python
if S.Sigma_moments is not None:
    ... 
```

Reply:
Yes, I agree. This has been introduced in the latest commit.